lib/tasks/migrate/20230331222716_create_tipline_newsletters.rake

```diff
@@ -0,0 +1,30 @@
+namespace :check do
+  namespace :migrate do
+    task create_tipline_newsletters: :environment do
+      TeamBotInstallation.where(user: BotUser.smooch_user).find_each do |tbi|
+        team = tbi.team
+        tbi.get_smooch_workflows.to_a.each do |workflow|
+          language = workflow['smooch_workflow_language']
+          unless workflow['smooch_newsletter'].blank?
+            newsletter = TiplineNewsletter.new
+            newsletter.introduction = workflow.dig('smooch_newsletter', 'smooch_newsletter_introduction')
+            newsletter.rss_feed_url = workflow.dig('smooch_newsletter', 'smooch_newsletter_feed_url')
+            articles = workflow.dig('smooch_newsletter', 'smooch_newsletter_body').to_s.split("\n\n")
+            newsletter.first_article = articles[0]
+            newsletter.second_article = articles[1]
+            newsletter.third_article = articles[2]
+            newsletter.number_of_articles = [[workflow.dig('smooch_newsletter', 'smooch_newsletter_number_of_articles').to_i, articles.size].max, 3].min
+            newsletter.send_every = workflow.dig('smooch_newsletter', 'smooch_newsletter_day')
+            newsletter.timezone = workflow.dig('smooch_newsletter', 'smooch_newsletter_timezone')
+            newsletter.time = Time.parse("#{workflow.dig('smooch_newsletter', 'smooch_newsletter_time')}:00")
+            newsletter.last_sent_at = workflow.dig('smooch_newsletter', 'smooch_newsletter_last_sent_at')
+            newsletter.language = language
+            newsletter.team = team
+            newsletter.save!
```

Comment:
couple of thoughts triggered by this save! method:
1. how can we make this idempotent, so that we don't accidentally create multiple newsletters
2. should we add a cache so that if an exception is raised that we don't restart from scratch, or are there few enough newsletters that it doesn't really matter?

Reply:
Good point! It actually made me realize that we don't need more than one newsletter for the same language for a given team. So I added a unique index for that case and it should avoid duplicates, make data more consistent and also make the rake task idempotent, because migrated newsletters will be ignored.